docs/user-tutorial/result-summary.md

```diff
@@ -0,0 +1,122 @@
+
+### rule file
+
+This section describes how to write rules in **rule file**.
+
+The convention is the same with [SuperBench Config File](https://microsoft.github.io/superbenchmark/docs/superbench-config), please view it first.
+
+Here is an overview of the rule file structure:
+
+scheme:
+```yaml
+version: string
+superbench:
+  rules:
+    ${rule_name}:
+      statistics:
+          - ${statistic_name}
+      categories: string
+      aggregate: (optional)[bool|string]
+      metrics:
+        - ${benchmark_name}/regex
+        - ${benchmark_name}/regex
+```
+
+example:
+```yaml
+# SuperBench rules
+version: v0.4
+superbench:
+  rules:
+    kernel_launch:
+      statistics:
+        - mean
+        - p90
+        - min
+        - max
+      aggregate: True
+      categories: KernelLaunch
+      metrics:
+        - kernel-launch/event_overhead
+        - kernel-launch/wall_overhead
+    nccl:
+      statistics: mean
+      categories: NCCL
+      metrics:
+        - nccl-bw/allreduce_8388608_busbw
+    ib-loopback:
+      statistics: mean
+      categories: RDMA
+      metrics:
+        - ib-loopback/IB_write_8388608_Avg_\d+
+      aggregate: ib-loopback/IB_write_.*_Avg_(\d+)
+```
+
+This rule file describes the rules used for the result summary.
+
+They are firstly organized by the rule name, and each rule mainly includes several elements:
+
+#### `metrics`
+
+The list of metrics for this rule. Each metric is in the format of ${benchmark_name}/regex, you can use regex after the first '/', but to be noticed, the benchmark name can not be a regex.
+
+#### `categories`
+
+User-defined category name in string belongs to the rule, which is used to classify and organize the metrics.
+
+#### `aggregate`
+
+This item is used to determine whether to aggregate the benchmark results from multiple devices to treat them as one collection.
+For example, aggregate the results of kernel-launch overhead from 8 GPU devices into one collection.
+
+The value of this item should be bool or pattern string with regex​:
+
+ - bool:
+    - `False`(default): if no aggregation.
+    - `True`: aggregate the results of multiple ranks. In detail, the metric names in `metrics' like 'metric:\\d+' will be aggregated and turned into 'metric' for most microbenchmark metrics.
+ - pattern string with regex: aggregate the results using the pattern string, which can be used to match the metric names in `metrics`. In detail, the part of the metric that matches the contents of () in the pattern string will be turned into *, other parts of the metric remain unchanged.
+
+#### `statistics`
+
+The list of statistical functions, which are used for this rule, to get the statistics of results from multiple nodes or multiple ranks.
+
+Statistical functions are supported currently:
+ - `mean`
+ - 'p{value}': {value} can be 1-99, for example, 'p50', 'p90', etc.
+ - 'min'
+ - 'max'
+ - 'std'
```

Comment:
Would you please change to alphabetical order?

Reply:
sure